### DISCUSSION TITLE
Function to help build tw classes more easily doesn't work

### DISCUSSION THREAD

**Discussion by Noriller** [tailwindlabs/tailwindcss#10053] [Help] [answered]
I was playing around with a utility function that would let me write tw classes more easily like:

```jsx
<div
  className={e(
    "my base classes",
    {
      sm: "classes for small breakpoint",
      md: ["classes for md", {
        hover: "with nested modifiers"
      }]
    }
  )}
>

// this creates:
<div class="my base classes sm:classes sm:for sm:small sm:breakpoint md:classes md:for md:md md:hover:with md:hover:nested md:hover:modifiers">
```

But it doesn't work. I then learned about how the default extractor works and as far as I saw, I can't just extend it or do something else...

Any way I can make this work? 
(I was thinking on making a package to use on any other project I have)

**Comment by wongjn** [comment, marked as answer]
You could use a [source transformer](https://tailwindcss.com/docs/content-configuration#transforming-source-files) to alter the source code that is read by Tailwind. You could write the transformer to somehow transform:
```jsx
<div
  className={e(
    "my base classes",
    {
      sm: "classes for small breakpoint",
      md: ["classes for md", {
        hover: "with nested modifiers"
      }]
    }
  )}
>
```
To something like:
```jsx
<div
  className={e(
    "my base classes",
    {
      sm: "sm:classes sm:for sm:small sm:breakpoint",
      md: ["md:classes md:for md:md", {
        hover: "md:hover:with md:hover:nested md:hover:modifiers"
      }]
    }
  )}
>
```
Note that this is only editing the string before being parsed by Tailwind to have the classes as full, unbroken strings and would not actually alter your source code.

**Reply by Noriller** [reply]
Awesome!

Do you know if there's an easier way to add transform to all file types or I have to name each one like:

```js
content: {
    files: ["./src/**/*.{js,ts,jsx,tsx}"],

    transform: {
      tsx: replacer,
      ts: replacer,
      jsx: replacer,
      js: replacer,
    }
  },
```

**Reply by wongjn** [reply]
I do not know if there's an easier way to add transform to all file types.

**Reply by RobinMalfait** [reply]
You can use DEFAULT as a key for a catch all.

**Reply by Noriller** [reply]
thank you!

**Reply by Noriller** [reply]
if anyone is interested, I'm giving it some final touches and then documenting:
https://www.npmjs.com/package/easy-tailwind

but you can start playing around here:
https://stackblitz.com/edit/easy-tailwind?file=src/App.jsx
